index.php

```diff
@@ -1,56 +1,61 @@
 <?php
-	mysql_connect ('localhost') ;
-	mysql_select_db ('test_db');
 
-	$sql = "SELECT * FROM php_blog ORDER BY timestamp DESC LIMIT 5";
-	$result = mysql_query($sql) or print ("Can't select entries from table php_blog.<br />" . $sql . "<br />" . mysql_error());
+	include('header.inc.php');
 
-	while($row = mysql_fetch_array($result))
+	$result = $database->query("SELECT * FROM `php_blog` ORDER BY `timestamp` DESC LIMIT 5");
+
+	foreach ($result as $row)
 	{
 		$date = date("D, d F Y, H:i ",$row['timestamp']);
 
-		$title = stripslashes($row['title']);
-		$entry = stripslashes($row['entry']);
+		$title = $row['title'];
+		$entry = $row['entry'];
 		$password = $row['password'];
 		$id = $row['id'];
 ?>
-		<p><strong><?php echo $title; ?></strong></p>
-	
-		<?php if($password == 1):?>
+		<p><strong><?=$title?></strong></p>
+
+<?php
+		if ( $password == 1 ) {
+?>
 			<p>This is a password protected entry. If you have a password, please log in below.</p>
-			<form method = "post" action = "view_post.php?id=<?php echo $id;?>">
+			<form method="post" action="/view_post.php?id=<?=$id?>">
 				<p>
 					<strong><label for="username">Username:</label></strong><br />
-					<input type = "text" name="username" id="username" />
+					<input name="username" id="username" />
 				</p>
 				<p>
 					<strong><label for="password">Password:</label></strong><br />
-					<input type = "password" name="pass" id="pass" />
+					<input type="password" name="pass" id="pass" />
 				</p>
 				<p>
 					<input type ="submit" name="submit" id="submit" value="submit" />
 				</p>
 			</form>
-		<?php else:?>
-				<p><?php echo $entry; ?></p><br>
-				<p>Posted on <?php echo $date;?></p>
-		<?php endif ;?>
+<?php
+		} else {
+?>
+				<p><?=nl2br($entry)?></p><br>
```

Comment:
Start doing this at render time rather than at database view time. A more realistic example would be if you were saving bbcode or markdown or something and realised your interpreter had a mistake. You'd want to be able to fix how it displayed for all posts and not just new ones. You can cache these results instead of doing them every time if theyre particularly resource intensive, but save the originals to the DB.